src/Elastic.Apm/Model/NoopSpan.cs

```diff
@@ -112,6 +112,8 @@ public Task<T> CaptureSpan<T>(string name, string type, Func<ISpan, Task<T>> fun
 
 		public void SetLabel(string key, decimal value) { }
 
+		public Label GetLabel(string key) => null;
```

Comment:
This is up for discussion. If the agent is turned off (`enabled=false` or `recording=false`), then this code runs.

My thinking was that if the agent is turned off, then it doesn't do anything and the advantage here is that we cause basically no overhead. If we new up `Context` instances and keep track of labels and other things, then the overhead with the turned off agent will be more.

I feel `NoopSpan` should be really NoOp and if the agent is turned off, we should work with NoOp instances.

Same for `NoopTransaction`.

Reply:
I think this is fine. The behaviour should be summarized in an `IMPORTANT` admonition in the enabled and recording configuration docs e.g. "When set to false. labels are not captured"